etc/st2packgen/st2packgen.py

```diff
@@ -40,73 +40,82 @@ def convert(name):
-        tmp['name'] = sname
-        tmp['type'] = stype
-        if 'documentation' in sdata:
-            tmp['description'] = striphtml(sdata['documentation'].rstrip().replace('"', "'"))
-        else:
-            tmp['description'] = ''
-
-        if sname in model.input_shape.required_members:
-            allvars['paramsreq'].append(tmp)
-        else:
-            allvars['params'].append(tmp)
-
-    actionyaml = outputdir + "/" + allvars['name'] + ".yaml"
-    y = open(actionyaml, 'w')
-
-    template = templateEnv.get_template('action_template.yaml.jinja')
-    outputText = template.render(allvars).encode('utf8')  # pylint: disable=no-member
-    y.write(outputText)
-
-    y.close()
+if myservice is None:
+    myservices = session.get_available_services()
+else:
+    myservices = [myservice]
+
+for myservice in myservices:
+    print "%s:" % myservice
+
+    try:
+        mysrv = session.get_service_model(myservice)
+    except botocore.exceptions.UnknownServiceError as e:
+        print "\n%s\n" % e
+        sys.exit(1)
+
+    for op in mysrv.operation_names:
+
+        allvars = {}
+        allvars['paramsreq'] = []
+        allvars['params'] = []
+
+        model = mysrv.operation_model(op)
+
+        op = convert(op)
+
+        print " " + op
+        allvars['action'] = op
+        allvars['name'] = myservice + "_" + op
+        allvars['service'] = myservice
+
+        if model.input_shape is None:
+            continue
+
+        members = model.input_shape.members
+
+        smodel = model.service_model
+
+        # print smodel._shape_resolver
+
+        smembers = model.input_shape._shape_model['members']
+        for sname, sdata in smembers.items():
+            tmp = {}
+            stype = smodel._shape_resolver._shape_map[sdata['shape']]['type']
+            # blob defined in boto as bytes or seekable file-like object - not supported here
+            if stype == "blob":
+                stype = "string"
+            if stype == "double":
+                stype = "number"
+            if stype == "long":
+                stype = "integer"
+            if stype == "structure":
+                stype = "object"
+            if stype == "map":
+                stype = "object"
+            if stype == "list":
+                stype = "array"
+            if stype == "timestamp":
+                stype = "string"
+            if stype == "float":
+                stype = "number"
+            tmp['name'] = sname
+            tmp['type'] = stype
+            if 'documentation' in sdata:
+                tmp['description'] = striphtml(sdata['documentation'].rstrip().replace("'", '"'))
+            else:
+                tmp['description'] = ''
+
+            if sname in model.input_shape.required_members:
+                allvars['paramsreq'].append(tmp)
+            else:
+                allvars['params'].append(tmp)
+
+        actionyaml = outputdir + "/" + allvars['name'] + ".yaml"
+        y = open(actionyaml, 'w')
```

Comment:
Minor thing - would be safer to use `with` to make sure file is always closed at the end.